File: numeric/max/max.jl

function max(values)
    current = 0
    for value in values
        if current < value
            current = value
        end
    end
    return current
end

list = [15, 10]
if length(ARGS) > 0
    list = [parse(Int, arg) for arg in ARGS]
end

maxValue = max(list)

println("values: ", list)
println("max: ", maxValue)

File: numeric/max/max.dart

int max(List<int> values) {
    int current = 0;
    for (var value in values) {
        if (value > current) {
            current = value;
        }
    }
    return current;
}

void main(List<String> arguments) {
    List<int> values = (arguments.length > 0) ?
        [for (var arg in arguments) int.parse(arg)]
        : [15, 10];

    int maxValue = max(values);

    print("values: ${values}");
    print("max: ${maxValue}");
}
